### ISSUE TITLE
Layout issues, especially in 5.1

### ISSUE THREAD

**Issue by Red-glyph** [HeinrichReimer/material-drawer#60]
I'm comparing the behaviour on Android 5.0.2 (with a real device Galaxy S4 I9505) and 5.1 (real device Nexus 7 2013 LTE).

The observations are related to the app in Google Play (v 2.1.1). Some of the problems I see in 5.1 are also observed by compiling the sources on 5.0.2 (a clone of today's repository), which is weird. I double-checked and re-installed the app from Play on 5.1 to make sure, but my conclusion is that those problems are the same I see by compiling the sources on a 5.0.2, not sure whether it's a regression.

1) Launching the application
=> similar appearance of the MainActivity's main fragment on both platforms.

2) Opening the navigation drawer by sliding.
=> completely blank on 5.1, contains 3 profiles and a few items on 5.0.2.
Sadly, the same pseudo-latin text is used everywhere, which makes it hard to link with the source and verify.
On 5.0.2, there are 7 items (including separators) and 2 fixed items.

3) Opening the navigation drawer with the ActionBar button
=> similar on both platforms

4) Rotating the screen
=> crashes on 5.0.2, OK on 5.1

5) Opening the app again, rotating the screen
=> OK

6) Selecting "DRAWERFRAMELAYOUT"
7) Opening the drawer (either by sliding or using the ActionBar)
=> repetition of the same pseudo-latin text on 5.0.2, completely blank on 5.1
Note: selecting an item (on 5.0.2) does not provide any feedback

8) Back to MainActivity and selecting "DRAWERACTIVITY"
=> 5.1: the text "Example of using ..." appears in the ActionBar (above "material-drawer" and the buttons, actually), the part below the ActionBar is empty
=> 5.0.2: the same text appears below, black on white

9) Opening the drawer (either by sliding or using the ActionBar)
=> similar on both, contains 2 profiles, and 2 items (one of which has a black flag). Selecting those items has no effect (except the ripple)

10) Selecting the other profile
=> changes the profile background and selects the 2nd profile

11) Selecting the other profile
=> fails to change the background, fails to select the profile. Now the two profiles have the same picture.

Note: that last part was also tested on a 4.x tablet, the same bug was observed.

In summary,
- layout issues in 5.0.2
- profile issues
- potential crash when the orientation changes
- needs more relevant text than this fake pseudo-latin repetition

**Comment by Red-glyph** [comment]
I also did some comparisons on a virtual device with 5.1 and it seems to misbehave in a similar way, though the animations are too slow for me to go on. It definitely needs some fixing anyway.

**Comment by Joadar** [comment]
Yes, I've the same problem about the number 2, my menu is blank and I'm on Android 4.4

**Comment by IgorGanapolsky** [comment]
Yes, I am experiencing issue number 2: sliding drawer with finger from left shows a blank drawer.  How to fix???

**Comment by HeinrichReimer** [comment]
Are you able to create a screen record? I don't really understand which type of bug this is.

**Comment by HeinrichReimer** [comment]
* **1:** What does this mean?
* **2-5:** Would be great if you record the screen.
* **7:** What does "completely blank" mean? It should only show a ripple effect, it shouldn't select any item.
* **8:** Happens because I changed the layout in 5.1 It should show a text inside the `Toolbar` to showcase custom `Toolbar` usage.
* **9:** This is just fine. See 7. (Didn't have time to generate a white "flag")
* **10:** This is fine.
* **11:** This is strange. It only happens if you select the profile directly (not from the profile list). I don't know why this happens though.

**Comment by iGio90** [comment]
I'm having the same issue. Seems like there are 2 drawers layout (one triggered by the toggle and one by the slide). The drawer layout that came with the slide is on top of the one triggered by the toggle

**Comment by troichet** [comment]
As iGio90, I have this problem. How to fix it? Your demo on google play is normal. but source file is not.

**Comment by iGio90** [comment]
I didn't checkout the library code yet. I'm gonna check it in an hour and pull the fix. Hope the owner of the project will merge it asap and push the update to gradle becouse I'm using the lib for a project and it fit everything I need. @HeinrichReimer i'll fix it later please merge once there is my pull!

**Comment by HeinrichReimer** [comment]
@iGio90 I'll do so

**Comment by HeinrichReimer** [comment]
@iGio90 any progress with your pull request?

**Comment by dimasolovyev** [comment]
The apk from play market works fine, though. But the demo project work as it described here: when i open navigation drawer with sliding from the side of the screen, it shows me blank drawer. But when i open the drawer with toolbar button or drawerLayout.openDrawer(drawer); it shows me correct drawer.
Again, the apk from play market shows correct drawer even if i open drawer with sliding from the side of the screen.

**Comment by rodrigokiller** [comment]
I am here just to mention that this problem occours with me too. 
Android 5.0.1
Nexus 5

**Comment by MeetRandy** [comment]
2) Opening the navigation drawer by sliding.
=> completely blank on 4.3, 4.4, 5.1

![screenshot_2015-06-24-09-24-03 1](https://cloud.githubusercontent.com/assets/13029168/8324651/eeb5bd54-1a52-11e5-977d-739c17333da5.png)
![screenshot_2015-06-24-09-24-18 1](https://cloud.githubusercontent.com/assets/13029168/8324654/f4cf5f42-1a52-11e5-897e-4cf99b59c6d9.png)

**Comment by MeetRandy** [comment]
guys i found a fix to the blank drawer when sliding, this is what i did: 
1.Remove (DrawerLayout drawerLayout = (DrawerLayout) findViewById(R.id.drawerLayout);
        drawer = (DrawerView) findViewById(R.id.drawer);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);)
2.each and every occurrence of "drawer" remove ie. (drawer.addItem), (drawer.selectItem) it should be just addItem, selectItem.
3. For a better understanding look at MainActivity3.java

**Comment by vivekvivi69** [comment]
thanx MeetRandy,, you saved my day... but still no round avatars on API 16.. thanks in advance sir HeinrichReimer..

**Comment by HeinrichReimer** [comment]
Simply change `DrawerActivity` to `AppCompatActivity` in *MainActivity.java* and *MainActivity2.java* :see_no_evil:

*PS:* That's the dark side of copy-paste :smile:
